packages/backend/src/managers/modelsManager.ts

```diff
@@ -85,7 +100,26 @@ export class ModelsManager {
     return path.resolve(this.#modelsDir, modelId, info.file);
   }
 
+  getLocalModelFolder(modelId: string): string {
+    return path.resolve(this.#modelsDir, modelId);
+  }
+
   getLocalModels(): LocalModelInfo[] {
     return Array.from(this.#localModels.values());
   }
+
+  async deleteLocalModel(modelId: string): Promise<void> {
+    const modelDir = this.getLocalModelFolder(modelId);
+    this.#deleted.add(modelId);
+    await this.sendModelsInfo();
+    try {
+      await fs.promises.rm(modelDir, { recursive: true });
```

Comment:
The recursive option for deleting always scares me, something we could do, is listing all the files inside the model directory, ask user for confirmation something like

"The following files will be deleted
- model-a.gguff
"

Then delete them by name instead of deleting the folder ? (I am maybe just overreacting)